tests/user.handles.test.ts

```diff
@@ -0,0 +1,169 @@
+    const handle = 'mycustomhandle';
+    const patchedUserResponse = await userController.patchCurrentUser({ user: { handle } }, member);
+
+    expect(patchedUserResponse.error).toBeNull();
+    const patchedUser = patchedUserResponse.user;
+    expect(patchedUser).toEqual({ ...member.getFullUserProfile(), handle });
+  });
+
+  test('User Cannot Update Their Handle to an already existing Handle', async () => {
+    const conn = await DatabaseConnection.get();
+    const [member, existingUser] = UserFactory.create(2);
+
+    await new PortalState().createUsers(member, existingUser).write();
+
+    const userController = ControllerFactory.user(conn);
+
+    const errorMessage = 'This handle is already in use.';
+    await expect(
+      userController.patchCurrentUser({ user: { handle: existingUser.handle } }, member),
+    ).rejects.toThrow(errorMessage);
+  });
+});
+
+describe('Get User Handles', () => {
+  test('User Can View Their Own Handle While Editing Profile', async () => {
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake();
+
+    await new PortalState().createUsers(member).write();
+
+    const userController = ControllerFactory.user(conn);
+    const userResponse = await userController.getCurrentUser(member);
+
+    expect(userResponse.error).toBeNull();
+    const { user } = userResponse;
+    expect(user.handle).toBe(member.handle);
+  });
+
+  test("Logged In User Can Fetch Another User's Profile by Handle", async () => {
+    const conn = await DatabaseConnection.get();
+    const [loggedInUser, otherUser] = UserFactory.create(2);
+
+    await new PortalState().createUsers(loggedInUser, otherUser).write();
+
+    const userController = ControllerFactory.user(conn);
+    const response = await userController.getUserByHandle(
+      { handle: otherUser.handle },
+      loggedInUser,
+    );
+
+    expect(response.user).toEqual(otherUser.getPublicProfile());
+  });
+});
+
+describe('Update to Invalid User Handle', () => {
+  test('Update handle error: too short', async () => {
+    const MIN_LENGTH = 3;
+    const user = UserFactory.fake({ handle: 'a' });
+
+    const errors = await validate(plainToClass(UserPatches, { ...user,
+      passwordChange: {
+        password: 'mypassword',
+        newPassword: 'mypassword',
+        confirmPassword: 'mypassword',
+      } }));
+
+    expect(errors).toBeDefined();
+    expect(errors).toHaveLength(1);
+    expect(errors[0].property).toEqual('handle');
+    expect(errors[0].constraints).toBeDefined();
+    expect(Object.keys(errors[0].constraints)).toHaveLength(1);
+    expect(errors[0].constraints.length).toBeDefined();
+    expect(errors[0].constraints.length)
+      .toBe(`handle must be longer than or equal to ${MIN_LENGTH} characters`);
+  });
+
+  test('Update handle error: too long', async () => {
+    const MAX_LEN = 32;
+    const user = UserFactory.fake({ handle: 'a'.repeat(2 * MAX_LEN) });
```

Comment:
nit: since we use faker conventionally anyways, let's just use faker here instead to match the other tests:
```suggestion
    const MAX_LEN = 32;
    const user = UserFactory.fake({ handle: faker.datatype.hexadecimal(2 * MAX_LENGTH) });
```
Introducing randomness like this is useful to make it clear that the only thing we care about controlling is the length, and not the actual string.